content/guides/use-case/wiremock.md

```diff
@@ -0,0 +1,293 @@
+
+
+## Using WireMock in development
+
+Now that you have tried WireMock, let’s use it in development and testing. In this example, you will use a sample application that has a Node.js backend. This app stack has the following configuration:
+
+  - Local Development Environment: The context in which the Node.js backend and WireMock are running.
+  - Node.js Backend: Represents the backend application that handles HTTP requests.
+  - External AccuWeather API: The real API from which live weather data is fetched.
+  - WireMock: The mock server that simulates the API responses during testing. It runs as a Docker container.
+
+  ![Diagram showing the architecture of WireMock in development ](./images/wiremock-arch.webp)
+
+  - In development, the Node.js backend sends a request to WireMock instead of the actual AccuWeather API.
+  - In production, it connects directly to the live AccuWeather API for real data.
+
+## Use mock data in local development
+
+Let’s set up a Node app to send requests to the WireMock container instead of the actual AccuWeather API.
+
+### Prerequisite
+
+- Install [Node and NPM](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)
+- Ensure that WireMock container is up and running. (as setup earlier).
+
+
+Follow the steps to setup a non-containerized Node application:
+
+1. Navigate to the `accuweather-api` directory
+
+   Make sure you're in the directory where your `package.json` file is located.
+
+2. Set the environment variable.
+
+   Open `.env` file placed under `accuweather-api/` directory. Remove the old entries and ensure that it just contains the following single line.
+ 
+   ```plaintext
+   API_ENDPOINT_BASE=http://localhost:8080
+   ```
+
+   This will tell your Node.js application to use the WireMock server for API calls. 
+
+3. Examine the Application Entry Point
+
+   - The main file for the application is `index.js`, located in the `accuweather-api/src/api` directory.
+   - This file starts the `getWeather.js` module, which is essential for your Node.js application. It uses the `dotenv` package to load environment variables from the`.env` file.
+   - Based on the value of `API_ENDPOINT_BASE`, the application routes requests either to the WireMock server (`http://localhost:8080`) or the AccuWeather API. In this setup, it uses the WireMock server.
+   - The code ensures that the `ACCUWEATHER_API_KEY` is required only if the application is not using WireMock, enhancing efficiency and avoiding errors.
+
+
+    ```javascript
+    require("dotenv").config();
+
+    const express = require("express");
+    const axios = require("axios");
+
+    const router = express.Router();
+    const API_ENDPOINT_BASE = process.env.API_ENDPOINT_BASE;
+    const API_KEY = process.env.ACCUWEATHER_API_KEY;
+
+    console.log('API_ENDPOINT_BASE:', API_ENDPOINT_BASE);  // Log after it's defined
+    console.log('ACCUWEATHER_API_KEY:', API_KEY);          // Log after it's defined
```

Comment:
We shouldn't recommend logging the API key

Reply:
@dvdksn Most of the application code is purposely kept simple to keep the focus on the guide's content and should not be considered production-ready. We also captured it in the form of statement under GitHub repo https://github.com/dockersamples/wiremock-node-docker?tab=readme-ov-file#project-overview 

But said that,  it's a best practice. Logging sensitive information like API keys can be a security risk. It’s important to avoid exposing such details in logs, especially in production environments. I can add that point as a tip. What do you think?

Reply:
Anyway, I went ahead and modified it to:

```
console.log('ACCUWEATHER_API_KEY is set:', !!API_KEY); //  i Log boolean instead of actual key
```

Now it doesn't log the keys:

```
API_ENDPOINT_BASE: http://dataservice.accuweather.com/
ACCUWEATHER_API_KEY is set: true
Listening: http://localhost:5000/
```